azurerm/internal/services/machinelearning/resource_arm_machine_learning_workspace.go

```diff
@@ -0,0 +1,425 @@
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.WorkspaceName,
+			},
+
+			"location": azure.SchemaLocation(),
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"application_insights_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				// TODO -- use the custom validation function of application insights
+				ValidateFunc: azure.ValidateResourceID,
+				// TODO -- remove when issue https://github.com/Azure/azure-rest-api-specs/issues/8323 is addressed
+				DiffSuppressFunc: suppress.CaseDifference,
+			},
+
+			"key_vault_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: keyVaultValidate.KeyVaultID,
+				// TODO -- remove when issue https://github.com/Azure/azure-rest-api-specs/issues/8323 is addressed
+				DiffSuppressFunc: suppress.CaseDifference,
+			},
+
+			"storage_account_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				// TODO -- use the custom validation function of storage account, when issue https://github.com/Azure/azure-rest-api-specs/issues/8323 is addressed
+				ValidateFunc: azure.ValidateResourceID,
+				// TODO -- remove when issue https://github.com/Azure/azure-rest-api-specs/issues/8323 is addressed
+				DiffSuppressFunc: suppress.CaseDifference,
+			},
+
+			"identity": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": {
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(machinelearningservices.SystemAssigned),
+							}, false),
+						},
+						"principal_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"tenant_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"container_registry_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				// TODO -- use the custom validation function of container registry
+				ValidateFunc: azure.ValidateResourceID,
+				// TODO -- remove when issue https://github.com/Azure/azure-rest-api-specs/issues/8323 is addressed
+				DiffSuppressFunc: suppress.CaseDifference,
+			},
+
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"friendly_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"sku_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: validation.StringInSlice([]string{
```

Comment:
could we put default just before the validation function?

Reply:
Done